### PR TITLE
Update documentation for missing params in header component

### LINKED ISSUE #3189: Missing params from documentation for header component
### Background/Issue

Some params aren't documented in the [Header component](https://service-manual.ons.gov.uk/design-system/components/header) for example the `subNavigation` ones. This needs documenting and we need to check if any others have been missed for the header.

### Tasks

1. Review the Header component and its variants and identify any missing parameters.
2. Document the missing parameters. 

### Acceptance Criteria

All missing parameters identified and subsequently documented.

### PR DESCRIPTION
**Pull request by precious-onyenaucheya-ons**
### What is the context of this PR?

Fixes #3189 
This PR updates the documentation of the header component to include parameters which were missing in the documentation.
### How to review this PR

Check that all parameters in the header component are well documented
### Checklist

This needs to be completed by the person raising the PR.

- [x] I have selected the correct Assignee
- [x] I have linked the correct Issue

### CODE REVIEW

**Review by alessioventuriniAND — APPROVED** [review]

File: src/components/header/_macro-options.md [hunk 1]
```diff
@@ -115,3 +117,14 @@
...
+| id                     | string                    | true     | The HTML `id` of the `<nav>` element. Used for the `aria-controls` attribute for the menu button displayed on small viewports. |
+| ariaLabel              | string                    | false    | The `aria-label` attribute added to the `<nav>` element. Defaults to “Main menu”.                                              |
+| itemsList              | array`<Item>`             | true     | Settings for an array of [list items](#item) for each navigation link                                                          |
+| currentPath            | string or array`<string>` | true     | The path of the current active page. Multiple paths can be provided using an array to highlight nested navigation.             |
+| toggleNavigationButton | array`<ToggleButton>`     | true     | Settings for the navigation [menu toggle button](#togglebutton) displayed on small viewports                                   |
+| removeHorizontalSubNav | boolean                   | false    | Set to “true” to remove the sub navigation                                                                                     |
```
Comment (rmccar) [inline src/components/header/_macro-options.md:130]: These two aren't used in subNavigation can you check all the others are used too and the descriptions are correct

File: src/components/header/_macro-options.md [hunk 2]
```diff
@@ -115,3 +117,14 @@
 | name       | string | false    | Sets the HTML `name` attribute for the `<button>`. Not valid if `url` is set.    |
 | url        | string | false    | If set, will create an HTML anchor link with the required classes and attributes |
 | attributes | object | false    | HTML attributes (for example, data attributes) to add to the button              |
+
+## SubNavigation
+
+| Name                   | Type                      | Required | Description                                                                                                                    |
+| ---------------------- | ------------------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------ |
+| id                     | string                    | true     | The HTML `id` of the `<nav>` element. Used for the `aria-controls` attribute for the menu button displayed on small viewports. |
+| ariaLabel              | string                    | false    | The `aria-label` attribute added to the `<nav>` element. Defaults to “Main menu”.                                              |
```
Comment (rmccar) [inline src/components/header/_macro-options.md:126]: This doesn't default to "Main menu" when using subNavigation", its "Section menu"

File: src/components/header/_macro-options.md (hunk 1 above)
Comment (precious-onyenaucheya-ons) [reply src/components/header/_macro-options.md:130]: I have removed the toggle button, but the remove horizontalSubNav is used

File: src/components/header/_macro-options.md (hunk 2 above)
Comment (precious-onyenaucheya-ons) [reply src/components/header/_macro-options.md]: updated

**Review by rmccar — APPROVED** [review]

**Review by SriHV — APPROVED** [review]

**Pull request merged**